### PR TITLE
Logger sanitizer

### PR DESCRIPTION
**Pull request by sar-perlman**
As @Jacfem discovered in the website logger testing, the sanitizer was missing strings which were potentially sensitive. Plus, I discovered that in our object handler, we only recursed if the value of the key was an object, and therefore, string-type values were being missed. All tests passed and should be all fixed.

### CODE REVIEW

**Comment by Jacfem** [comment]
Agree with @farleyta's thoughts there! But confirming that I tested this locally with the website, and that now Honeybadger will not send sensitive keys that exist as strings inside of the message key.

**Comment by sar-perlman** [comment]
@Jacfem @farleyta  refactored to explicitly handle array, string and non-null object, then return all others. Also added a unit test to make sure `Should not redact a number, undefined, null, or boolean`.

File: logger-sanitizer/index.js [hunk 1]
```diff
@@ -26,16 +23,21 @@ const loggerSanitizer = (data) => {
...
+      } else {
+        acc[key] = loggerSanitizer(data[key], blacklistedLogKeys);
+      }
+      /* eslint-enable no-param-reassign */
+      return acc;
+    }, {});
+  }
+
+  // is another type like boolean or number, return as-is
+  return data;
```
Comment (farleyta) [inline logger-sanitizer/index.js]: Initially, we explicitly checked for `!_isPlainObject` _above_ the object handling code, and inside this block, returned `data` as-is.  Then as the base case, because all other types are handled explicitly above, we could be confident that executing the `Object.keys` logic would happen on a true Object.  I _think_ the above code does the same thing, but what happens with newer constructs like `Set` or `Map`? or `Date`?

In my quick tests, these will get picked up by `typeof(data) === 'object'` – but will return empty arrays during `Object.keys(data)`, essentially being bypassed without explicitly being redacted.  If we make the switch I mention above, they'd get returned as-is (therefore, no redaction), which I don't think we want either.

What do you think about explicitly handling edge-case non-Object constructs that still match `typeof(data) === "object"` above the return on line 27?

Something like:

```js
if (typeof(data)==='object' && data !== null) {
  const objectKeys = Object.keys(data);

  // account for non-plain objects (Map, Set, Date, etc)

  // account for "normal" objects :)
  if (objectKeys && objectKeys.length === 0) return '[redacted]';
  return objectKeys.reduce(
    // ...same as before
  );
}
```

Comment (sar-perlman) [reply logger-sanitizer/index.js]: Good points @farleyta In what situations do you think we would get a `set` or `map` pushed to logs?

Comment (farleyta) [reply logger-sanitizer/index.js]: Great question, @sar-perlman – I can't think of a single situation where this would _intentionally_ happen, which is why this is probably "nice to have".  But you never know – if  a Set or Map gets passed unintentionally as a parameter and then logged (due to error or general info/notification) we may want to ensure we're handling.  I think this is very edge case (hence the suggestion to go straight to `[redacted]` and not put in the effort to attempt to inspect and only sanitize as needed).
